src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -950,12 +975,9 @@ private void verifyMacroAwareTranscode(
         }
         if (streamType == StreamType.TEXT) {
             String rewritten = out.toString(StandardCharsets.UTF_8.name());
-            assertEquals(expectedNumberOfIonVersionMarkers, countOccurrencesOfSubstring(rewritten, "$ion_1_1"));
-            assertEquals(expectedNumberOfAddSymbolsInvocations, countOccurrencesOfSubstring(rewritten, SystemSymbols_1_1.ADD_SYMBOLS.getText()));
-            assertEquals(expectedNumberOfAddMacrosInvocations, countOccurrencesOfSubstring(rewritten, SystemSymbols_1_1.ADD_MACROS.getText()));
-            assertEquals(expectedNumberOfSetSymbolsInvocations, countOccurrencesOfSubstring(rewritten, SystemSymbols_1_1.SET_SYMBOLS.getText()));
-            assertEquals(expectedNumberOfSetMacrosInvocations, countOccurrencesOfSubstring(rewritten, SystemSymbols_1_1.SET_MACROS.getText()));
-            assertEquals(expectedNumberOfExplicitEncodingDirectives, countOccurrencesOfSubstring(rewritten, SystemSymbols_1_1.ION_ENCODING.getText()));
+            for (Matcher<String> expectation : expectations) {
+                assertThat(rewritten, expectation);
+            }
```

Comment:
My comment made me go look at the AllOf implementation, where I realized that there is already a distinction between `describeTo` and `describeMismatch`, which `AllOf` uses to enumerate only the failing matches. I just had to read more of the failure message. Given that, I suggest this instead:

```suggestion
            assertThat(rewritten, allOf(expectation));
```

This will also require the appropriate `import static org.hamcrest.Matchers.allOf;` in the imports section. For a failing test, it will generate a message like this:
 
```plain
Expected: (a String including 1 occurrences of $ion_1_1 and a String including 1 occurrences of add_symbols and a String including 0 occurrences of add_macros and a String including 0 occurrences of set_symbols and a String including 0 occurrences of set_macros and a String including 2 occurrences of $ion_encoding)
     but: a String including 1 occurrences of add_symbols was "$ion_1_1 $ion_encoding::((symbol_table [\"SimonSays\",\"anything\"])) $ion_encoding::((symbol_table [\"foo\"]) (macro_table (macro SimonSays (anything) (%anything)))) (:SimonSays [(:SimonSays {$1:1.23e0}),(:SimonSays 123),\"abc\"]) [] "
```

The 'x and y and z and... ' part is not super helpful, but the `but: a String including 1 occurrences of add_symbols was ...` part is. YMMV.

Reply:
Done, thanks